File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/TestStartEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import com.testerum.runner.events.model.position.EventKey
import java.time.LocalDateTime

data class TestStartEvent @JsonCreator constructor(
        @JsonProperty("time")         override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("eventKey")     override val eventKey: EventKey,
        @JsonProperty("testName")     val testName: String,
        @JsonProperty("testFilePath") val testFilePath: Path
): RunnerEvent

File: web/web-frontend/src/app/model/test/event/test-start.event.ts
import {RunnerEvent} from "./runner.event";
import {Path} from "../../infrastructure/path/path.model";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";

export class TestStartEvent implements RunnerEvent, Serializable<TestStartEvent> {
    time: Date;
    eventKey: string;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.TEST_START_EVENT;

    testName: string;
    testFilePath: Path;
    tags: Array<string> = [];

    deserialize(input: Object): TestStartEvent {
        this.eventKey = input["eventKey"];
        this.time = new Date(input["time"]);
        this.testName = input["testName"];
        this.testFilePath = Path.deserialize(input["testFilePath"]);
        this.tags = input['tags'] || [];

        return this;
    }

    serialize(): string {
        return undefined;
    }
}
